internal/common/db.go

```diff
@@ -390,21 +390,65 @@ func GetHostCountByRole(db *gorm.DB, clusterID strfmt.UUID, role models.HostRole
 	return &count, nil
 }
 
-func prepareClusterDB(db *gorm.DB, eagerLoading EagerLoadingState, includeDeleted DeleteRecordsState, conditions ...interface{}) *gorm.DB {
+func prepareClusterDBWithJoins(db *gorm.DB, eagerLoading EagerLoadingState, includeDeleted DeleteRecordsState, conditions ...interface{}) *gorm.DB {
 	if includeDeleted {
 		db = db.Unscoped()
 	}
 
-	conditions = append(conditions, func(db *gorm.DB) *gorm.DB {
-		if includeDeleted {
-			return db.Unscoped()
-		}
+	if !eagerLoading {
 		return db
-	})
+	}
 
-	if eagerLoading {
-		for _, tableName := range ClusterSubTables {
-			db = LoadTableFromDB(db, tableName, conditions...)
+	tableConfigs := []struct {
+		association string
+		tableName   string
+	}{
+		{MonitoredOperatorsTable, "monitored_operators"},
+		{ClusterNetworksTable, "cluster_networks"},
+		{ServiceNetworksTable, "service_networks"},
+		{MachineNetworksTable, "machine_networks"},
+		{APIVIPsTable, "api_vips"},
+		{IngressVIPsTable, "ingress_vips"},
+	}
+
+	if !includeDeleted {
+		db = db.Preload(HostsTable, func(db *gorm.DB) *gorm.DB {
+			preloadDB := db.Joins("INNER JOIN clusters ON hosts.cluster_id = clusters.id").
+				Where("hosts.deleted_at IS NULL AND clusters.deleted_at IS NULL")
```

Comment:
And the end it's a single query, right? Then I don't think the need to repeat that for each of the tables